lib/db/set.go

```diff
@@ -380,6 +382,27 @@ func (s *Snapshot) RemoteNeedFolderFiles(device protocol.DeviceID, page, perpage
 	return files
 }
 
+func (s *Snapshot) WithPathsMatchingBlocksHash(hash []byte, iterator StringIterator) error {
+	key, err := s.t.keyer.GenerateBlockMapListKey(nil, []byte(s.folder), hash, nil)
+	if err != nil {
+		return err
+	}
+	iter, err := s.t.NewPrefixIterator(key)
+	if err != nil {
+		return err
+	}
+
+	for iter.Next() {
+		file := string(s.t.keyer.NameFromBlockMapListKey(iter.Key()))
+		if !iterator(file) {
+			break
+		}
+	}
+
+	iter.Release()
```

Comment:
I'd defer it. Not necessary here but is just safer e.g. in case an early return gets added later.

Reply:
Addressed.